pkg/apis/elasticsearch/v1alpha1/types.go

```diff
@@ -56,6 +57,28 @@ type ElasticsearchNode struct {
 	Storage      ElasticsearchNodeStorageSource `json:"storage"`
 }
 
+type ElasticsearchStorageSpec struct {
+	LocalVolume *ElasticsearchLocalVolumeSpec `json:"localVolume,omitempty"`
+	StorageClass *ElasticsearchStorageClassSpec `json:"storageClass,omitempty"`
+
+	// PersistentVolumeClaim will NOT try to regenerate PVC, it will be used
+	// as-is. You may want to use it instead of VolumeClaimTemplate in case
+	// you already have bounded PersistentVolumeClaims you want to use, and the names
+	// of these PersistentVolumeClaims doesn't follow the naming convention.
+	PersistentVolumeClaim *v1.PersistentVolumeClaimVolumeSource `json:"persistentVolumeClaim,omitempty"`
+}
+
+type ElasticsearchLocalVolumeSpec struct {
+	Path				string `json:"path"`
+	Size				resource.Quantity `json:"size"`
+}
+
+type ElasticsearchStorageClassSpec struct {
+	StorageClassName				string `json:"storageClassName"`
```

Comment:
Maybe this should be `name` since its already nested under `storageClass`

Reply:
This is to match the PVC spec field name